src/components/Table/index.tsx

```diff
@@ -0,0 +1,123 @@
+import { ColumnDef, flexRender, getCoreRowModel, useReactTable } from '@tanstack/react-table'
+import { useLayoutEffect, useRef, useState } from 'react'
+import styled from 'styled-components'
+
+const Wrapper = styled.div`
+  border: 1px solid ${({ theme }) => theme.surface3};
+  border-radius: 20px;
+  overflow: hidden;
+`
+const TableContainer = styled.div`
+  overflow: auto; // Enable horizontal scroll on full table
+`
+const TableHeadContainer = styled.div`
+  // Enable horizontal scroll on the header but hide scrollbar
+  overflow-x: auto;
+  ::-webkit-scrollbar {
+    display: none;
+  }
+  -ms-overflow-style: none;
+  scrollbar-width: none;
+`
+const TableBodyContainer = styled.div`
+  max-height: 550px;
+  overflow-y: auto; // Enable vertical scroll only on table body
+`
+const TableSection = styled.table`
+  border-collapse: collapse;
```

Comment:
this also needs `width: 100%;` to fit parent container
For example when this table is used elsewhere
<img width="1728" alt="Screenshot 2023-11-09 at 11 51 16 AM" src="https://github.com/Uniswap/interface/assets/11512321/eb1b2910-a9ac-4062-b659-b1730ce5e9a0">
<img width="1728" alt="Screenshot 2023-11-09 at 11 51 30 AM" src="https://github.com/Uniswap/interface/assets/11512321/9a6f5286-318f-4756-bd60-9aad57ec3a52">

Reply:
this actually runs into an issue where the `TableCells` offsetWidth is too large and basically always gets an overflow